index.js

```
@@ -1511,6 +1519,64 @@
+
+/**
+ * Middleware that restricts access to admin-only routes by verifying the user is authenticated and matches the configured admin user ID.
+ * Redirects unauthenticated users to the Discord login page and denies access to non-admin users.
+ */
+function ensureAdmin(req, res, next) {
+  if (req.isAuthenticated && req.isAuthenticated() && req.user && req.user.id === process.env.ADMIN_USER_ID) {
+    return next();
+  }
+  // If not admin, redirect to login or show an error
+  if (req.user) { // User is logged in but not admin
+    res.status(403).send("Access Denied. You are not authorized to view this page. <a href='/logout'>Logout</a>");
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)